### PR TITLE
github: use issue forms, link to discussions, remove discord

### PR DESCRIPTION
**Pull request by katexochen**
### Checklist

- [x] Add labels (e.g., for changelog category)

### CODE REVIEW

**Comment by katexochen** [comment]
Unsure, if there is a way to preview this on a branch. :/

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 blank_issues_enabled: true
 contact_links:
-  - name: Confidential Computing Discord
-    url: https://discord.com/invite/rH8QTH56JN
-    about: Join the Confidential Computing community!
+  - name: Having trouble or questions?
+    url: https://github.com/edgelesssys/constellation/discussions/new?category=q-a
```
Comment (derpsteb) [inline .github/ISSUE_TEMPLATE/config.yml:4]: I guess this should be discussed with @thomasten and potentially marketing.

**Review by derpsteb — APPROVED** [review]
Once the change from Discord has been discussed: :+1:

**Comment by 3u13r** [comment]
LGTM, defering approval to Biz/Marketing regarding Discord.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: Feature request
+description: Suggest an idea for this project
+body:
+  - type: textarea
+    attributes:
+      label: Use case
+      description: A concise description of what you want to happen.
+    validations:
+      required: true
```
[the remark this answers is not in the document]
Comment (katexochen) [reply .github/ISSUE_TEMPLATE/feature_request.yml:9]: Like I said, I did not. Did you check in a private repo?

> Note: The required field key is only supported in public repositories. In private and internal repositories, all fields are optional.

[the remark this answers is not in the document]
Comment (katexochen) [reply .github/ISSUE_TEMPLATE/feature_request.yml:9]: I guess we would need to test in a public repo, but I guess we could just merge and see if its working. Won't hurt to much if it not for an hour.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  - type: textarea
+    attributes:
+      label: Describe your solution
+      description: In case you have an idea how to implement this feature.
+  - type: checkboxes
+    attributes:
+      label: Would you be willing to implement this feature?
+      options:
+        - label: Yes, I could contribute this feature.
+          required: true
```
Comment (thomasten) [inline .github/ISSUE_TEMPLATE/feature_request.yml:19]:
```suggestion
```
I guess we also want to accept feature requests without the willingness to implement

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    attributes:
+      label: Constellation Config
+      description: Make sure to remove sensitive information, e.g., `yq e 'del(.provider.*.project)' constellation-conf.yaml`.
+      placeholder: |
+        <details>
+        <summary>Content of `constellation-conf.yaml`</summary>
+        ```yaml
+        Your config goes here
+        ```
+        </details>
```
[the remark this answers is not in the document]
Comment (katexochen) [reply .github/ISSUE_TEMPLATE/bug_report.yml:32]: That's not copied, I wrote that from scratch. It enables collapsing the config. But we can also remove if you don't like it.

[the remark this answers is not in the document]
Comment (thomasten) [reply .github/ISSUE_TEMPLATE/bug_report.yml:32]: Sorry, I somehow thought I saw that in the deleted bug_report.md ...
However, it doesn't work as you think https://github.com/thomasten/test1/issues/new?assignees=&labels=bug%2Cneeds-triage&projects=&template=bug_report.yml
So you could, e.g., use the first 2-3 lines of a constellation conf file for this

Comment (katexochen) [inline .github/ISSUE_TEMPLATE/bug_report.yml:32]:
```suggestion
```

**Review by thomasten — APPROVED** [review]

**Pull request merged**